pkg/sources/gitlab/gitlab.go

```diff
@@ -368,7 +383,11 @@ func (s *Source) getAllProjects(ctx context.Context, apiClient *gitlab.Client) (
 	for {
 		groupList, res, err := apiClient.Groups.ListGroups(&listGroupsOptions)
 		if err != nil {
-			return nil, fmt.Errorf("received error on listing groups, you probably don't have permissions to do that: %w", err)
+			err = fmt.Errorf("received error on listing groups, you probably don't have permissions to do that: %w", err)
+			if err := reportErr(ctx, err); err != nil {
+				return nil, err
+			}
+			break
```

Comment:
same question about `break`

Reply:
Yep, this was intentional. Do you think it's okay to do this? I'll double check the flow.